# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dhcp6c/hotfix.patch
# Copyright (C) 2019 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- dhcp6c-20190419/dhcp6c.c.vanilla	2019-08-08 12:38:07.110760347 +0200
+++ dhcp6c-20190419/dhcp6c.c	2019-08-08 12:39:56.086762471 +0200
@@ -55,6 +55,8 @@
 #include <netinet6/in6_var.h>
 #endif
 
+#include <linux/ipv6.h>
+
 #include <arpa/inet.h>
 #include <netdb.h>
 
--- dhcp6c-20190419/auth.c.vanilla	2019-08-08 12:40:06.114762666 +0200
+++ dhcp6c-20190419/auth.c	2019-08-08 12:40:19.686762931 +0200
@@ -119,7 +119,7 @@
 }
 
 int
-dhcp6_calc_mac(char *buf, size_t len, int proto __unused, int alg,
+dhcp6_calc_mac(char *buf, size_t len, int proto, int alg,
     size_t off, struct keyinfo *key)
 {
 	hmacmd5_t ctx;
@@ -148,7 +148,7 @@
 }
 
 int
-dhcp6_verify_mac(char *buf, ssize_t len, int proto __unused,
+dhcp6_verify_mac(char *buf, ssize_t len, int proto,
     int alg, size_t off, struct keyinfo *key)
 {
 	hmacmd5_t ctx;
